plugins/Installation/Controller.php

```diff
@@ -698,41 +698,33 @@ private function redirectToNextStep($currentStep, $parameters = array())
      *
      * @return string|false
      */
-    private function extractHost($url)
+    private function extractHostAndPort($url)
     {
-        $urlParts = parse_url($url);
-        if (isset($urlParts['host']) && strlen($host = $urlParts['host'])) {
-            return $host;
+        $host = parse_url($url, PHP_URL_HOST) ?? false;
+
+        if (empty($host)) {
+            return false;
+        }
+
+        $port = (int) parse_url($url, PHP_URL_PORT) ?? 0;
+
+        if (!empty($port) && $port !== 80 && $port !== 443) {
+            return $host . ':' . $port;
         }
 
-        return false;
+        return $host;
     }
 
     /**
-     * Add trusted hosts
+     * Sets trusted hosts in config
      */
-    private function addTrustedHosts($siteUrl)
+    private function setTrustedHost(Config $config): void
     {
-        $trustedHosts = array();
-
-        // extract host from the request header
-        if (($host = $this->extractHost('http://' . Url::getHost())) !== false) {
-            $trustedHosts[] = $host;
-        }
-
-        // extract host from first web site
-        if (($host = $this->extractHost(urldecode($siteUrl))) !== false) {
-            $trustedHosts[] = $host;
-        }
```

Comment:
@tsteur Do you know why we used to set the host of the site added during installation to the trusted host list? We are not adding trusted hosts when adding additional sites through UI or API, nor would we ever update it if the url of that site would be changed, so I wonder if there was a good reason for it.

Reply:
@sgiehl actually noticed the same problem yesterday when I set up a 5.x system. I can't think of any reason. Seems it's been in the code base for a very long time. It may be fine to remove it?